service/src/main/java/com/codeforcommunity/processor/ProcessorImpl.java

```diff
@@ -81,18 +147,92 @@ public boolean addMember(String first, String last) {
   }
 
   @Override
-  public boolean validate(String first, String last) {
-    Result result = db.fetch(
-        "select first_name, last_name\n" + "   from member\n" + "   where first_name = ? and last_name = ?;", first,
-        last);
-    if (result.isEmpty())
+  public UserReturn getUserByEmail(String email) {
+
+    try {
+
+      Result userResult = db.fetch("select * from users where email=?;", email);
+      if (userResult.isEmpty()) {
+        return null;
```

Comment:
As another general rule, never return null from a method. Ideally you can refactor this to have a standard NoUser type of object but if not you should at least be using an `Optional<UserReturn>` as your return type.